File: internship_backend_nov24/jury_solutions/python/c.py
size, border, changes = map(int, input().split())
elements = list(map(int, input().split()))
left = 0
right = 0
currentChanges = 0
maxLength = 0
while (right < size):
    if elements[right] < border:
        right += 1
        continue
    if currentChanges < changes:
        currentChanges += 1
        right += 1
        continue
    maxLength = max(maxLength, right - left)

    while elements[left] < border:
        left += 1

    left += 1
    right += 1
maxLength = max(maxLength, right - left)
print(maxLength)

File: internship_backend_nov24/jury_solutions/cs/c.cs
using System;
using System.Linq;


class Program
{
    static void Main()
    {
        Solve();
    }

    static void Solve()
    {
        var input = Console.ReadLine().Split().Select(int.Parse).ToArray();
        var size = input[0];
        var border = input[1];
        var changes = input[2];
        var elements = Console.ReadLine().Split().Select(int.Parse).ToArray();
        var left = 0;
        var right = 0;
        var currentChanges = 0;
        var maxLength = 0;
        while (right < size)
        {
            if (elements[right] < border)
            {
                right++;
                continue;
            }

            if (currentChanges < changes)
            {
                currentChanges++;
                right++;
                continue;
            }

            maxLength = Math.Max(maxLength, right - left);

            while (elements[left] < border)
                left++;
        
        
            left++;
            right++;
        }

        maxLength = Math.Max(maxLength, right - left);

        Console.WriteLine(maxLength);
    }
}